### DISCUSSION TITLE
Pimcore update 6.9.x to 10.x failed

### DISCUSSION THREAD

**Discussion by Hbitvof** [pimcore/pimcore#9582] [General]
Hi,

I follow the instructions on https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/Updating_Pimcore/V6_to_V10.html but I am unable to update because of the following error:

`Script Pimcore\Composer::clearCache handling the pimcore-scripts event terminated with an exception

  [RuntimeException]
  An error occurred when executing the "/usr/bin/php --php-ini=/etc/php.ini bin/console --ansi cache:clear --no-warmup" command:

  Exit code: 255

  mcore
   *
   * This source file is available under two different licenses:
   * - GNU General Public License version 3 (GPLv3)
   * - Pimcore Enterprise License (PEL)
   * Full copyright and license information is available in
   * LICENSE.md which is distributed with this source code.
   *
   * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
   * @license    http://www.pimcore.org/license     GPLv3 and PEL
   */

  namespace App;

  use HWI\Bundle\OAuthBundle\HWIOAuthBundle;
  use Pimcore\HttpKernel\BundleCollection\BundleCollection;
  use Pimcore\Kernel as PimcoreKernel;

  class App\Kernel extends PimcoreKernel
  {
      /**
       * Adds bundles to register to the bundle collection. The collection is able
       * to handle priorities and environment specific bundles.
       *
       * @param BundleCollection $collection
       */
      public function registerBundlesToCollection(BundleCollection $collection)
      {

          // activate bundle for SSO oauth login/register functionality
          if (class_exists('\Http\HttplugBundle\HttplugBundle')) {
              $collection->addBundle( new \Http\HttplugBundle\HttplugBundle());
          }
          $collection->addBundle(HWIOAuthBundle::class);

      }
  }}

  RuntimeException {#5
    #message: "The autoloader expected class "App\Kernel" to be defined in file "/var/www/html/pimcore/vendor/composer/../../src/Kernel.php". The file was found but the class was not in it, the class name or namespace probably has a typo."
    #code: 0
    #file: "./vendor/symfony/symfony/src/Symfony/Component/ErrorHandler/DebugClassLoader.php"
    #line: 401
    trace: {
      ./vendor/symfony/symfony/src/Symfony/Component/ErrorHandler/DebugClassLoader.php:401 { …}
      ./vendor/symfony/symfony/src/Symfony/Component/ErrorHandler/DebugClassLoader.php:358 { …}
      Symfony\Component\ErrorHandler\DebugClassLoader->loadClass() {}
      ./vendor/pimcore/pimcore/lib/Bootstrap.php:246 { …}
      ./vendor/pimcore/pimcore/lib/Bootstrap.php:60 { …}
      ./bin/console:24 {
        › /** @var \Pimcore\Kernel $kernel */
        › $kernel = \Pimcore\Bootstrap::startupCli();
        › $application = new \Pimcore\Console\Application($kernel);
      }
    }
  }

  2021-06-26T16:58:19+00:00 [critical] Uncaught Exception: The autoloader expected class "App\Kernel" to be defined in file "/var/www/html/pimcore/vendor/composer/../../src/Kernel.php". The file was found but the class was not in it, the class name or namespace probab
  ly has a typo.
`
The file Kernel.php is from https://raw.githubusercontent.com/pimcore/demo/10.x/src/Kernel.php

What's going wrong here?

**Comment by dvesh3** [comment]
@Hbitvof please make sure, if you have correctly adapted changes on composer.json https://github.com/pimcore/demo/blob/10.x/composer.json#L28

**Reply by Hbitvof** [reply]
@dvesh3 I did, I use https://raw.githubusercontent.com/pimcore/demo/10.x/composer.json as file but still same result.

**Comment by ShaileshT-dot** [comment]
Hi, In the file src/Kernel.php, can you try changing the name of the class from App\Kernel to Kernel?
